src/generate/utils/State_AI.cpp

```diff
@@ -48,4 +52,123 @@ namespace spy::gameplay {
 
         return leafSuccessors;
     }
+
+    void State_AI::addDamage(const spy::character::Character &character, int damage) {
```

Comment:
kann man nicht direkt auf das set an characteren zugreifen und damage hinzufügen?

Reply:
Ja, aber dann muss ich in der evaluation function immer den state bevor der Charakter was gemacht hat mit dem state nachdem der Charakter mit seinem Zug fertig ist vergleichen und ein diff bilden. So schreibe ich mir quasi das diff in den execute methoden direkt in den state kann in der eval Funktion auf diese Werte zugreifen.